changelog/922.improvement.md

```diff
@@ -0,0 +1,31 @@
+## Problem
+Rasa knowledge base actions cannot infer the object type of an object directly from the user message. This prevents action_query_knowledge from providing a suitable response when a user asks for a certain attribute of an object unless the user first asks to list objects related to that object type even though the knowledge base has the relevant information. That is, the knowledge base actions require the slot `object_type` to be set to one of the primary key values in the knowledge base for it to search through the objects. Here is an example:
```

Comment:
Why is the changelog changed again here?

Reply:
The change is in the function name Thiru. In changelong I have referenced the function `get_object_type_dynamic` which is then renamed in this PR. So I changed the changelog here. But I will reflect this change in the changelog PR too. To avoid conflicts.

Reply:
@Thirunayan22 , if this is okay we can merge this